python/cudnn/hstu_attention/_kernels/mask.py

```
@@ -0,0 +1,442 @@
+        for i in cutlass.range(cute.size(preds), unroll_full=True):
+            preds[i] = True
+
+        for i in cutlass.range(cute.size(preds), unroll_full=True):
+            block_row = cute.get(tScS_t2r[i], mode=[row_id])
+            row = block_row + base_row
+            q_row = row - seqlen_offset
+
+            block_col = cute.get(tScS_t2r[i], mode=[col_id])
+            col = block_col + base_col
+
+            if col >= self.seqlen_k or row >= self.seqlen_q + seqlen_offset and mask_seqlen:
```

Ruff (0.16.2)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)